### DISCUSSION TITLE
<DynChild> created for MaybeSignal::Static

### DISCUSSION THREAD

**Discussion by smessmer** [leptos-rs/leptos#1732] [Q&A]
I'm using `#[prop(into)] prop: MaybeSignal<...>` heavily in all of my components, to allow call sites to choose whether to pass in a signal or a reactive value. I'm using this even for props for which a call site will most likely choose static values, e.g. styling, color, button text, ...

My assumption is that, because i use `MaybeSignal` and not `Signal`, there is no performance downside to using it instead of taking a prop by value, so I can just as well make everything `MaybeSignal` and give the call site flexibility, even if it's not needed.

However, I just noticed that there is actually a difference in the generated HTML. `MaybeSignal`, if added to a `view!{}`, seems to create html comments with `<DynChild>` tags around its content, even if it's `MaybeSignal::Static`.

I'm not fully sure what `<DynChild>` does and if it has any impact on hydration or performance. But even if it doesn't, it would probably be cleaner to not generate those for `MaybeSignal::Static` and only generate them for `MaybeSignal::Dynamic`.

**Comment by gbj** [comment]
You don't provide any example code, unfortunately. (And could you please use the issue template, or open a Discussion, in the future?)

Your assumption is incorrect, depending on how you are reading from `MaybeSignal`, which is why an example or reproduction would be useful in answering the question. Reading `MaybeSignal` inside a function will treat it like any other signal being read inside a function, generating DynChild. Matching on the `MaybeSignal` and rendering it either reactively or not will not.

For what it's worth you're also looking at the output in debug mode, which is intentionally verbose (marking boundaries around different components with comment markers.)
